tests/NuGet.IndexingTests/PackageAnalyzerTests.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Indexing;
+using NuGet.IndexingTests.TestSupport;
+using Xunit;
+
+namespace NuGet.IndexingTests
+{
+    public class PackageAnalyzerTests
+    {
+        [Theory, MemberData("TheoryData")]
+        public void AddsCorrectFieldAnalyzers(string field, string text, TokenAttributes[] expected)
```

Comment:
just a question really, but structuring the tests like this as an enumeration rather than separate tests - isn't that harder to isolate results - the whole thing is reported as one shot rather than one part failing - isn't that harder to reason over?